### ISSUE TITLE
LL Connection failure with MAC OS + Broadcom BCM_4387

### ISSUE THREAD

**Issue by sacca97** [whad-team/whad-client#59]
Using the example peripheral device. This is what the Wireshark capture looks like from WHAD:

<img width="1036" alt="image" src="https://github.com/user-attachments/assets/04cf354e-f14e-4bab-827a-0a744afe11a9">

Different behavior with external dongle (e.g., realtek) or with BCM_4378.

**Comment by virtualabs** [comment]
I know this behavior, it happened to me once. The BLE stack is supposed to send a single LL_VERSION_IND PDU but for some reason it reacts each time it receives an LL_VERSION_PDU by ... sending another  LL_VERSION_PDU. Fun part is that the peripheral answers each PDU while the specs definitely states it must only answer the first request. 

I have no clue why this is happening, for now. And yes, the BCM stack is also weird :rofl:.

**Comment by sacca97** [comment]
`    def is_version_sent(self, conn_handle):
        if conn_handle in self.connections:
            self.connections[conn_handle]["version_sent"]
        return False`
        
I think i see the problem of the double response by the peripheral btw @virtualabs . Gonna check out properly tomorrow when I have the devices at hand.

**Comment by sacca97** [comment]
Yup that is indeed the problem as method always return false. However it does not change the final outcome

![image](https://github.com/user-attachments/assets/f6e2c0ff-3382-4b39-872c-2c86b74487ab)

**Comment by sacca97** [comment]
Additionally the Arrival Time is obv wrong, and it could be that one causing sync problems, but I don't know how to fix it, yet
